### PR TITLE
Updates instructions & expected outcomes for redaction sample

### PR DESCRIPTION
**Pull request by psx95**
### Issue - Minor updates to redaction sample

Minor documentation updates required for redaction sample.

### Fix
 - Updates instructions to be in line with the simple sample. 
 - Adds expected outcomes section to clarify the intent of the sample. 
 - **Also updates the `build/local/Makefile` target -  `docker-build` to update the collector image location for redaction sample as well**

### Testing
  Followed the instructions as a standalone sample (meaning no prior run for simple sample) and managed to get the expected outcome.

### CODE REVIEW

File: build/local/Makefile [hunk 1]
```diff
@@ -31,6 +31,7 @@ build: setup
 docker-build: setup
 	docker build -t ${IMAGE_NAME}:${IMAGE_VERSION} .
 	sed -i "s/%OTEL_COLLECTOR_IMAGE%/${REGISTRY_LOCATION}-docker.pkg.dev\/${GCLOUD_PROJECT}\/${CONTAINER_REGISTRY}\/${IMAGE_NAME}:${IMAGE_VERSION}/g" ../../deploy/gke/simple/manifest.yaml
+	sed -i "s/%OTEL_COLLECTOR_IMAGE%/${REGISTRY_LOCATION}-docker.pkg.dev\/${GCLOUD_PROJECT}\/${CONTAINER_REGISTRY}\/${IMAGE_NAME}:${IMAGE_VERSION}/g" ../../deploy/gke/redaction/manifest.yaml
```
Comment (damemi) [inline build/local/Makefile:34]: this is good for now, but we should ultimately figure out a way to abstract this so we don't need a new line for every sample (that can be its own PR)

**Review by damemi — APPROVED** [review]

File: build/local/Makefile (hunk 1 above)
Comment (psx95) [reply build/local/Makefile:34]: I see, makes sense. I initially thought adding a line in the Readme about how to edit this file, but found that approach to be potentially confusing.  

What should our intent be here - 
 -  Do we need a better way to achieve what this build target is doing ? 
  OR
 -  Do we need to find an alternative to this approach altogether ? (Something along the lines to find a way to specify what sample to build the image for)

Comment (damemi) [reply build/local/Makefile:34]: A better (more generic) way to do what this build target is doing. For example, something like a common target that each sample's Makefile imports and the sample sets which files to update is the kind of approach I'm thinking of

Comment (psx95) [reply build/local/Makefile:34]: This gives me an idea, if we apply the sed command in each sample's makefile and ask the user to run that target before deploying the sample, then that would also fix this problem. 
(The individual sample's makefile could also refer to some targets in build directory's makefiles to achieve this) 

A good side benefit is that in case the collector image is already built and pushed to Artifact Registry, the users would not have to `make docker-build` again just to update the manifest files with the correct names. This also kind of decouples the `build`  and `deploy` directories since running commands from the build directory would not edit the files in deploy.
